osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -316,7 +323,7 @@ private void processBeatmapChanges()
             {
                 foreach (var set in setsRequiringRemoval) removeBeatmapSet(set);
 
-                foreach (var set in setsRequiringUpdate) updateBeatmapSet(set);
+                foreach (var set in setsRequiringUpdate) updateBeatmapSet(fetchFromID(set)!);
```

Comment:
Why does the `removeBeatmapSet()` call above use `set` raw, but this uses `fetchFromID(set)`? What's the refetch for anyway?

Reply:
The methods take different arguments. One is a `Guid` and one is a `BeatmapSet`.